### ISSUE TITLE
Undefined behavior warnings in gms_graphics_compare_luminance()

### ISSUE THREAD

**Issue by angstsmurf** [DavidKinder/Magnetic#2]
When running _The Pawn_ in Gargoyle,  Undefined Behavior Sanitizer complains about load of misaligned address at lines 1004 and 1005 in glk.c:
https://github.com/DavidKinder/Magnetic/blob/1e789037244ea19f1053da88e984df68f255c67d/Glk/glk.c#L1004-L1005

`Load of misaligned address 0x0001062ed084 for type 'long', which requires 8 byte alignment`

**Comment by DavidKinder** [comment]
Looking at the code, I think this is a fairly simple mistake in gms_graphics_compare_luminance(), which will only show up when sizeof(long) > sizeof(int), which occurs for 64-bit Linux/OSX.

Would you try an experiment for me? Change gms_graphics_compare_luminance() to be
````
static int
gms_graphics_compare_luminance (const void *void_first, const void *void_second)
{
  int first = *(int *) void_first;
  int second = *(int *) void_second;

  return first > second ? 1 : second > first ? -1 : 0;
}
````
Check that 1) the graphics still work, and 2) the misaligned addresses messages go away.

**Comment by angstsmurf** [comment]
Yes, graphics work and Undefined Behavior Sanitizer is happy.

**Comment by DavidKinder** [comment]
Fixed by commit 2ac9fce58f8f38158f9d6cc6c9d3f02b882e738d. Apologies for taking so long to get round to looking at this!
